request/request.go

```diff
@@ -252,16 +252,16 @@ func (r *Request) Scrub(reply *dns.Msg) *dns.Msg {
 	// Account for the OPT record that gets added in SizeAndDo(), subtract that length.
 	sub := 0
 	if r.Req.IsEdns0() != nil {
-		sub = optLen
+		size -= optLen
+		sub = 1
 	}
 
 	// subtract to make spaces for re-added EDNS0 OPT RR.
 	re := len(reply.Extra) - sub
-	size -= sub
 
 	l, m := 0, 0
 	origExtra := reply.Extra
-	for l < re {
+	for l <= re {
```

Comment:
this warrants a comment to why `<=` needs to be used here

Reply:
This is the standard way to implement a binary search. If we exit the loop when `l == re`, then we potentially aren't checking one of the indices in the slice. We might be, because of integer division, but there's no guarantee we are.